src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorPage.cs

```diff
@@ -119,6 +120,36 @@ public dynamic ThemeLayout
             }
         }
 
+        public string ViewLayout
+        {
+            get
+            {
+                if (ThemeLayout is IShape layout)
+                {
+                    if (layout.Metadata.Alternates.Count > 0)
+                    {
+                        return layout.Metadata.Alternates.First();
+                    }
+
+                    return layout.Metadata.Type;
+                }
+
+                return String.Empty;
+            }
+
+            set
+            {
+                if (ThemeLayout is IShape layout)
+                {
+                    var alternates = layout.Metadata.Alternates.ToList();
```

Comment:
Too many new collection to insert an element. If the `AlternatesColletion` doesn't have a way to insert an item, let's add it.

Reply:
Okay, i will do.

Reply:
Done